src/pages/datasets/index.tsx

```diff
@@ -1,74 +1,124 @@
-import type {FunctionComponent} from 'react'
+import {FunctionComponent, useState} from 'react'
+import {useQuery} from 'react-query'
 import {DatasetCard} from '@/components/pages/datasets/cards/datasets'
 import {ArrowForward} from '@/components/icons/arrows'
+import {fetchDatasets, fetchRequests} from '@/pages/api/datasets'
+import type {IDataset, IRequest} from '@/types/datasets'
+import {Plus} from '@/components/icons/plus'
+import {SearchBar} from '@/components/forms/searchbar'
+import Link from 'next/link'
 
 const Datasets: FunctionComponent = () => {
+  const {isLoading, data: datasetsData, error} = useQuery<IDataset[], Error>('datasets', fetchDatasets)
```

Comment:
`useQuery` never identifies that `fetchDatasets` has failed. Thus, `error` is always `null`.

The practical consequence is that If the `fetchDatasets` call fails, the app would crash as `datasetsData === undefined`